### PR TITLE
executeConfigFile: use require('vm').executeInNewContext

### PR DESCRIPTION
**Pull request by fictitious**
as discussed in   
https://github.com/systemjs/builder/pull/394#discussion_r44321335

### CODE REVIEW

File: lib/builder.js [hunk 1]
```diff
@@ -275,8 +275,7 @@ function executeConfigFile(saveForReset, ignoreBaseURL, source) {
     systemConfigFunc(cfg); 
   };
 
-  // jshint evil:true
-  eval(source.toString());
+  require('vm').runInNewContext(source.toString(), {System: System, console: console});
```
Comment (guybedford) [inline lib/builder.js]: I had a thought here. How about doing:

```javascript
var g = Object.create(global);
g.System = System;
```

That way we provide access to all globals, but deny writes.

**Comment by guybedford** [comment]
Thanks for the follow-up!

**Comment by fictitious** [comment]
Turns out it's slightly more convoluted. This is what I ended up with after doing some more tests.

**Comment by guybedford** [comment]
Just to confirm I'm following correctly - `Object.create(global)` as the context fails for the `global.x = 'asfd'` case?

**Comment by fictitious** [comment]
Yes - after

    var p = Object.create(global)

the following returns true:

    p.global === global

also, in node, there are also p.GLOBAL and p.root aliases referring to the same global. The code makes all these aliases refer to new context instead.

**Comment by guybedford** [comment]
@fictitious since we know the global names in Node are `root`, `global`, `GLOBAL`, I'd prefer to just do this manually to avoid complete traversal of the global.

**Comment by fictitious** [comment]
ok - assuming the risk of it ever changing in node is small enough

**Comment by guybedford** [comment]
:+1:

**Pull request merged**